src/sidebar/util/term-before-position.ts

```diff
@@ -1,9 +1,43 @@
 /**
  * Returns the "word" right before a specific position in an input string.
  *
- * In this context, a word is anything between a space or newline, and provided
- * position.
+ * In this context, a word is anything between a space, newline or tab, and
+ * provided position.
  */
 export function termBeforePosition(text: string, position: number): string {
   return text.slice(0, position).match(/\S+$/)?.[0] ?? '';
 }
+
+export type TermCoordinates = {
+  start: number;
+  end: number;
+};
+
+/**
+ * Returns the `start` and `end` positions for the term that overlaps with
+ * provided reference position.
+ *
+ * For example, given the text "hello hypothesis", and the reference position 9
+ * (which corresponds to the `p` character) it will return the start and end of
+ * `hypothesis`, hence { start: 6, end: 16 }.
+ *
+ * Useful to get the coordinates of the word matching the caret in text inputs
+ * and textareas.
+ */
+export function getTermCoordinatesAtPosition(
```

Comment:
I think it makes sense for this function to live next to `termBeforePosition` (it even proved to be useful to reuse part of the existing test), but the module then probably needs to be renamed.